libs/providers/flagd/src/lib/service/in-process/grpc/grpc-fetch.ts

```diff
@@ -70,39 +70,67 @@ export class GrpcFetch implements DataFetch {
   ) {
     this._logger?.debug('Starting gRPC sync connection');
     closeStreamIfDefined(this._syncStream);
-    this._syncStream = this._syncClient.syncFlags(this._request);
```

Comment:
This can throw before it ever connects... when that happens the `.on('error')` handler never takes effect. This is fixed by making the catch and `.on('error')` both share a local method (`handleError`).